src/topology/README.md

```diff
@@ -0,0 +1,150 @@
+interface-topology
+========================
+
+> Implementation of the topology interface used by the `js-libp2p` registrar.
+
+This interface has two main purposes. It uniforms the registration of libp2p protocols and enables a smarter connection management.
```

Comment:
I think something like this would be good for now, and we can point to a more in depth overview once that's ready.

```suggestion
Topologies can be used in conjunction with `js-libp2p` to help shape its network and the overlays of its subsystems, such as pubsub and the DHT.
```